rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/SecureCxfClientFactoryBuilder.java

```diff
@@ -0,0 +1,323 @@
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+package org.codice.ddf.cxf;
+
+import ddf.security.PropertiesLoader;
+import ddf.security.service.SecurityServiceException;
+import ddf.security.settings.SecuritySettingsService;
+import ddf.security.sts.client.configuration.STSClientConfiguration;
+import org.apache.commons.lang.StringUtils;
+import org.apache.cxf.Bus;
+import org.apache.cxf.BusException;
+import org.apache.cxf.common.util.CollectionUtils;
+import org.apache.cxf.configuration.jsse.TLSClientParameters;
+import org.apache.cxf.endpoint.EndpointException;
+import org.apache.cxf.interceptor.LoggingInInterceptor;
+import org.apache.cxf.interceptor.LoggingOutInterceptor;
+import org.apache.cxf.jaxrs.client.Client;
+import org.apache.cxf.jaxrs.client.ClientConfiguration;
+import org.apache.cxf.jaxrs.client.JAXRSClientFactoryBean;
+import org.apache.cxf.jaxrs.client.WebClient;
+import org.apache.cxf.transport.http.HTTPConduit;
+import org.apache.cxf.transport.https.HttpsURLConnectionFactory;
+import org.apache.cxf.ws.security.SecurityConstants;
+import org.apache.cxf.ws.security.tokenstore.SecurityToken;
+import org.apache.cxf.ws.security.trust.STSClient;
+import org.apache.shiro.subject.Subject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Element;
+
+import javax.ws.rs.core.Cookie;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+public class SecureCxfClientFactoryBuilder {
+
+    protected static final String ADDRESSING_NAMESPACE = "http://www.w3.org/2005/08/addressing";
+
+    private static final transient Logger LOGGER = LoggerFactory
+            .getLogger(SecureCxfClientFactoryBuilder.class);
+
+    private final SecuritySettingsService securitySettingsService;
+
+    private final STSClientConfiguration stsClientConfig;
+
+    /**
+     * Blueprint constructor.
+     */
+    public SecureCxfClientFactoryBuilder(SecuritySettingsService securitySettingsService,
+            STSClientConfiguration stsClientConfig) throws SecurityServiceException {
+        if (securitySettingsService == null || stsClientConfig == null) {
+            throw new SecurityServiceException("Could not access security configurations.");
+        }
+        this.securitySettingsService = securitySettingsService;
+        this.stsClientConfig = stsClientConfig;
+    }
+
+    /**
+     * @see SecureCxfClientFactoryBuilder#buildFactory(String, Class, String, String, java.util.List, boolean)
+     */
+    public <T> SecureCxfClientFactory<T> buildFactory(String endpointUrl, Class<T> interfaceClass)
```

Comment:
My intent is not to discourage ongoing feedback; you've already provided some useful comments. I'd like to think this will be merge-worthy in a few more days, but that will only happen if people keep commenting on it.

I talked to @pklinef about this a bit, and we're trying something a little different here. I deliberately made the PR early so that people could see and comment on the design before it became too inflexible and difficult to change.